.github/workflows/check.yml

```diff
@@ -88,16 +92,17 @@ jobs:
     steps:
       - name: Let us have colors
         run: echo "FORCE_COLOR=true" >> "$GITHUB_ENV"
-      - uses: actions/setup-python@v5
-        with:
-          python-version: "3.12"
       - name: Install the latest version of uv
         uses: astral-sh/setup-uv@v2
         with:
           enable-cache: true
           github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup python hatch
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
       - name: install hatch
-        run: uv pip install hatch --system
+        run: uv pip install --system --python 3.12 hatch
```

Comment:
Isn't this redundant? I don't see the purpose.

Reply:
Nevermind, I misinterpreted the previous code block. I thought it was this https://hatch.pypa.io/latest/install/#github-actions

I do wonder if that's faster.

Reply:
The UI rounds the UV version to zero and the official Hatch action to one. Behind the scenes action downloads a prebuilt distribution so I suppose an optimization, exclusively for GitHub Actions, would be to download just `site-packages`. I'm not going to do this because that's a lot of work for one second but just thought I would take this out in case my stream of consciousness is useful to anyone watching 😆

Reply:
Without this installs it into python3.10 that had some permission issues https://github.com/tox-dev/platformdirs/actions/runs/10835509905/job/30131356681?pr=301